config/data_sources/covid_tracking_project.tf

```diff
@@ -5,3 +5,20 @@ resource "google_bigquery_dataset" "bq_covid_tracking_project" {
   dataset_id = "covid_tracking_project"
   location   = "US"
 }
+
+# We need to define the metadata table in Terraform so that we can grant
+# access to the gcs_to_bq runner to make queries to it.
+resource "google_bigquery_table" "covid_tracking_project_metadata" {
+  dataset_id = google_bigquery_dataset.bq_covid_tracking_project.dataset_id
+  table_id = var.ctp_metadata_table_name
+}
+
+resource "google_bigquery_table_iam_binding" "covid_tracking_project_metadata_iam_binding" {
```

Comment:
For my understanding - why is this necessary?

If this is a pattern specifically required for some datasets based on whether they have to use a metadata table, maybe add this to the doc explaining how to add a new dataset?

Reply:
In order for a user (or service account) to run query jobs, they need to have read access to the table. The way our project is set up, our tables are locked down on creation, so the service accounts running our Cloud Run services don't have access. So we need to grant them access by defining the table and its permissions in terraform. When testing on my cloud project, I wasn't able to get the gcs_to_bq service to query the metadata table unless this iam binding was defined.

This will be applicable to any of our Cloud Run services that wish to query any BQ table (e.g. if we wanted to join with county/state fips codes during standardization in a similar way). I'm currently writing documentation for this :) Also thinking about how we might allow new dependencies without having to re-create the table in terraform...

Reply:
Hmm, I think I ran into the a similar issue with the aggregator, and got it working by adding the `bigquery.tables.getData` role. It looks like the gcs_to_bq service doesn't have that permission - maybe that would fix it?